### PR TITLE
Documenting H1-FPN messages

### PR DESCRIPTION
**Pull request by makrsmark**

### CODE REVIEW

**Comment by kevinelliott** [comment]
@makrsmark You're back at it! Pretty soon I'm going to have to give you the #airframes-dev role if you keep it up at this rate.

Looks good to me.

One thing I'm thinking might be nice would be a section for all these pages that indicate some kind of parsing strategy (even if just a single paragraph), towards the end. For example, perhaps to describe the parse/decode flow, and then provide generic pseudocode and/or any regexes.

For you for this I'm not considering it a requirement, but if you feel like taking a stab at doing it to help define the general format, could be very helpful!

**Comment by makrsmark** [comment]
@kevinelliott I don't think there's much past splitting on `:` and grouping the pairs... except for the route/waypoint stuff... which i haven't figured out yet. I'm looking at updating the typescript decoder... we'll see how that goes!

File: research/H1/FPN.md
Comment (makrsmark) [inline research/H1/FPN.md:23]: in case it's not obvious - the `(?)` is to indicate i'm not 100% sure

Comment (kevinelliott) [reply research/H1/FPN.md:23]: Total acceptable. If anyone else (or myself) identifies, we can update later.

**Comment by kevinelliott** [comment]
> @kevinelliott I don't think there's much past splitting on `:` and grouping the pairs... except for the route/waypoint stuff... which i haven't figured out yet. I'm looking at updating the typescript decoder... we'll see how that goes!

Which is all that would really need to be stated in the new section. But some formalization might be useful because then non-coders could play along and think things through.

**Comment by makrsmark** [comment]
Will do!

**Comment by kevinelliott** [comment]
> Will do!

Awesome! Then I'll merge when you ping me here again. Really loving the additions. Will watch for a PR in the other repo too.

**Comment by kevinelliott** [comment]
I've added @andermatt64 to review as well, just to start keeping him in the loop as he's been working on this area a lot with me and has been spearheading a rust-based decoder library too.

File: research/H1/FPN.md
Comment (andermatt64) [inline research/H1/FPN.md:91]: `RP` may be Route Planned?

File: research/H1/FPN.md
Comment (andermatt64) [inline research/H1/FPN.md:56]: typo

File: research/H1/FPN.md
Comment (andermatt64) [inline research/H1/FPN.md:34]: For the route, it looks like it may be something like `[NAVAID/WAYPOINT],[GPS DDM]` delimited by `.`. The GPS appears to be Degree Decimal Minute format (somewhat confirmed if you plug in the navaid/waypoint into [opennav](https://opennav.com) and look at the coordinates).

I have another example (assuming in the format) that shows that the delimiter is `.` and not `..`:
<pre>
FPN/FNAAL2338/RP:DA:KPVD:AA:KCLT:CR:PVDCLT01:R:05O(36R):A:CHSLY5.LYH:AP:ILS36R..MXE,N39551W075402.Q75.GVE..LYH,N37153W079142A557
</pre>
Notice `MXE,N39551W075402.Q75.GVE`, this is a `[VORTAC],[GPS COORDS].[ROUTE].[NEXT VORTAC]`. However, it is odd that there is no GPS for the `NEXT VORTAC`

File: research/H1/FPN.md
Comment (andermatt64) [inline research/H1/FPN.md:58]: Do we want to further extract the arrival procedure + transition point?

**Review by andermatt64 — APPROVED** [review]
Awesome work @makrsmark ! I made a few comments.

It's really hard to parse out some of these formats, there's quite a few variants and the documentation is basically non-existent. Always ends up being a grey-to-black box reverse engineering job heh.

File: research/H1/FPN.md
Comment (makrsmark) [reply research/H1/FPN.md:56]: lysdexia got the best of me -will fix to `KDFW`

File: research/H1/FPN.md
Comment (makrsmark) [reply research/H1/FPN.md:58]: yes, I can extract it into `{procedureName},{transitionPoint}`

File: research/H1/FPN.md
Comment (makrsmark) [reply research/H1/FPN.md:91]: yes! i dropped that by accident - will update the doc

File: research/H1/FPN.md
Comment (makrsmark) [reply research/H1/FPN.md:34]: Yea, i was thinking it might be`..{waypoint}[,{position}]` but the coords seemed too far off. I guess just to make sure, when you say DDM `N39551W075402` would convert to `39.551,-075.402`, correct?

I'll add this message to the docs, which scraps the waypoint name and just goes from position to position

```
FPN/RI:DA:KMEM:AA:EGSS:F:HUMMS,N36214W088591..PXV,N37557W087457..ROD,N
40173W084026..SLT,N41308W077582.J190..RKA,N42280W075144..MIILS,N46524W
067029.N279A..TUDEP,N51100W053140..N52000W050000..N53000W040000..N5400
0W030000..N54000W020000..DOGAL,N54000W015000..BEXET,N54000W014000..BAK
UR,N52145W005408.UN546..STU,N51597W005024.UP2..NUMPO,N51366W003170.Y3.
.BEDEK,N51223W001335CE78
```

`..` vs `.` really bothers me... i'd love to figure out what it means

File: research/H1/FPN.md
Comment (makrsmark) [reply research/H1/FPN.md:58]: I'm going to hold off on this, but update the docs, i think all procedures can be in the format `{name}.{waypoints}` where waypoints is the same format as routes

**Comment by kevinelliott** [comment]
@makrsmark ready for merge?

**Comment by makrsmark** [comment]
> @makrsmark ready for merge?

now should be ok

File: research/H1/FPN.md
Comment (andermatt64) [reply research/H1/FPN.md:44]: So from your example, let's take `ROD`. The GPS coords per OpenNav is  40° 17' 16.06" N, 84° 2' 35.31" W (see https://opennav.com/navaid/US/ROD/ROSEWOOD ). It sort matches the `N40173W084026` after it: `N 40 17.3, W 84 02.6`. 

Regarding, `..` and `.`, the same example unfortunately just busted through my assumptions with `STU,N51597W005024.UP2..NUMPO,N51366W003170`...

**Pull request merged**
